src/quote/indexQuoteProvider.ts

```diff
@@ -122,6 +131,48 @@ export class FlashMintQuoteProvider
           tx,
         }
       }
+      case FlashMintContractType.leveragedExtended: {
+        if (!zeroExApiV1) {
+          throw new Error('Contract type requires ZeroExApiV1 to be defined')
+        }
+        const leverageExtendedQuoteProvider =
+          new LeveragedExtendedQuoteProvider(provider, zeroExApiV1)
+        const leveragedExtendedQuote =
+          await leverageExtendedQuoteProvider.getQuote(request)
+        if (!leveragedExtendedQuote) return null
+        const builder = new LeveragedExtendedTransactionBuilder(provider)
+        const txRequest: FlashMintLeveragedExtendedBuildRequest = {
+          isMinting,
+          inputToken: inputToken.address,
+          inputTokenSymbol: inputToken.symbol,
+          outputToken: outputToken.address,
+          outputTokenSymbol: outputToken.symbol,
+          inputTokenAmount: BigNumber.from(0), // TODO: leveragedExtendedQuote.inputTokenAmount,
+          outputTokenAmount: BigNumber.from(0), // TODO: leveragedExtendedQuote.outputTokenAmount,
+          swapDataDebtCollateral: leveragedExtendedQuote.swapDataDebtCollateral,
+          swapDataInputOutputToken: leveragedExtendedQuote.swapDataPaymentToken,
+          swapDataInputTokenForETH:
+            leveragedExtendedQuote.swapDataDebtCollateral, // FIXME: add swapDataInputTokenForETH (returned from quote provider)
```

Comment:
Please check with @ckoopmann how to construct this swap data - as I understood it's only relevant for ERC20's and can be ignored for ETH as input token.